actionpack/lib/action_dispatch/journey/router.rb

```diff
@@ -122,7 +122,7 @@ def find_routes(req)
           routes.map! { |r|
             match_data = r.path.match(req.path_info)
             path_parameters = {}
-            match_data.names.zip(match_data.captures) { |name, val|
+            match_data.named_captures.each { |name, val|
```

Comment:
Would this decrease the performance of this method now that we are calling `to_h` and `each`?